ADR/0015-integration-service-two-phase-architecture.md

```diff
@@ -0,0 +1,94 @@
+# 15. The Two-phase Architecture of the Integration Service
+
+Date Documented: 2023-02-09
+
+## Status
+
+In review
+
+## Context
+
+The Integration Service is in charge of running integration test pipelines by executing
+the Tekton pipeline for each user-defined IntegrationTestScenario.
+The main payload that’s being tested are Snapshots which contain references to all
+Components that belong to the Application, along with their images.
+
+A problem we face: Since testing an application with Tekton pipelines takes time,
+and multiple component builds can happen in quick succession, there is potential
+for race conditions between different Snapshots that are created for each of those builds.
+This would primarily manifest itself by two Snapshots getting promoted in quick
+succession, but neither of them having the latest images contained within them.
```

Comment:
```suggestion
One problem faced by the integration service is caused by the fact that testing an
application with Tekton pipelines takes time. During the course of testing an application, 
multiple component builds can happen in quick succession, leading to a potential race
condition between different Snapshots that are created for each of those builds.
This would primarily manifest itself by two Snapshots getting promoted in quick
succession but neither of them having the latest images contained within them.
```

Reply:
Ack, updated with the suggestion.